knack/config.py

```diff
@@ -141,14 +141,21 @@ def set_value(self, section, option, value):
     def set_to_use_local_config(self, use_local_config):
         self.use_local_config = use_local_config
 
+    def remove_option(self, section, option):
+        for config in self._config_file_chain if self.use_local_config else self._config_file_chain[-1:]:
+            if config.remove_option(section, option):
+                return True
```

Comment:
Not sure if it is a good idea to fall back to the parent config if `option` is not found. The other way is to only remove the option in this local config and keep the behavior same as `set_value`. 

Switch `use_local_config` like [`ConfiguredDefaultSetter`](https://github.com/Azure/azure-cli/blob/34f9033b96e6ea31f82626a9f2b152c1a061300e/src/azure-cli/azure/cli/command_modules/configure/custom.py#L150) does, or provide a scope parameter in `remove_option` to override `self.use_local_config`.

Again, not sure. I am open to this.

Reply:
It's a little different with `set_value`. I think `remove` may need map to `get`. As we can list values of parent config, I think it's reasonable to remove them in the same file.